scylla/src/transport/topology.rs

```diff
@@ -251,12 +251,21 @@ async fn query_peers(conn: &Connection, connect_port: u16) -> Result<Vec<Peer>,
         let address = SocketAddr::new(ip_address, connect_port);
 
         // Parse string representation of tokens as integer values
-        let tokens: Vec<Token> = tokens_str
+        let tokens: Vec<Token> = match tokens_str
             .iter()
             .map(|s| Token::from_str(s))
             .collect::<Result<Vec<Token>, _>>()
-            .map_err(|_| QueryError::ProtocolError("Couldn't parse tokens as integer values"))?;
-
+        {
+            Ok(parsed) => parsed,
+            Err(e) => {
+                // FIXME: we could allow the users to provide custom partitioning information
+                // in order for it to work with non-standard token sizes.
+                // Also, we could implement support for Cassandra's other standard partitioners
+                // like RandomPartitioner or ByteOrderedPartitioner.
+                warn!("Couldn't parse tokens as 64-bit integers: {}, proceeding with a dummy token. If you're using a partitioner with different token size, consider migrating to murmur3", e);
+                vec![Token { value: 0 }]
```

Comment:
Another option for choosing the token value: `(i * maxuint64 / N)`, where `i` is the index of the node and `N` is the number of the nodes. Not sure which is better.

Reply:
Random sounds good, it'll have more consistent behavior when topology changes

Reply:
Btw - for amazon keyspaces it wouldn't matter a bit, because they expose a single endpoint, but until we implement RandomPartitioner and friends it's indeed better to avoid using a single token. (It's even better if the clients avoid token aware load balancing without murmur)